src/lando/main/scm/abstract_scm.py

```diff
@@ -102,6 +94,29 @@ def apply_patch(
             None
         """
 
+    @abstractmethod
+    def process_merge_conflict(
+        self,
+        exception: PatchConflict,
+        pull_path: str,
+        revision_id: int,
+    ) -> dict[str, Any]:
+        """Process merge conflict information captured in a PatchConflict, and return a
+        parsed structure.
+
+        The structure is a nested dict as follows:
+
+            revision_id: revision_id
+            failed paths: list[dict]
+                path: str
+                url: str
+                changeset_id: str
+            rejects_paths: dict[str, dict]
+                <str>: dict[str, str] (conflicted file path)
+                    path: str (reject file path)
+                    content: str
```

Comment:
Nit: this looks ripe for a `dataclass` implementation. ;)

Reply:
It... does doesn't it? I walked away from it due to the nestedness of the structure, and it being slightly off-track, but it would be nice.

https://bugzilla.mozilla.org/show_bug.cgi?id=1941245